### DISCUSSION TITLE
Allow top bottom leading trailing widgets for bubble

### DISCUSSION THREAD

**Discussion by alan2030** [flyerhq/flutter_chat_ui#653] [Ideas & Requests]
Hi! I was wondering whether this library supports chat actions? For example one user makes an offer and another user is prompted with "Accept" or "Decline" flat buttons in chat. Something similar to zendesk's bot navigation feature.

**Comment by henry2man** [comment]
What about `customBottomWidget`? Probably you can do something with this ;)

From the code: 

```dart
  /// Allows you to replace the default Input widget e.g. if you want to create
  /// a channel view.
  final Widget? customBottomWidget;
```

**Comment by demchenkoalex** [comment]
`customBottomWidget` is something appearing instead of the `Input`, like said in the documentation comment, for the channel view for example, where you can't reply. Original question I think is about buttons below messages? People actually achieved that using `bubbleBuilder` in #301, but there is no "official" way. Second time seeing this though, maybe should be added natively.

**Comment by henry2man** [comment]
I've used this to create some similar (predefined answer buttons for a support chat).

I've achieved this by adding a Column with custom actions in one row and next the input box, which is logically also needed. 

But it could be great (and probably easy) to have something native.

**Comment by demchenkoalex** [comment]
so you meant buttons on top of an input in the end? ok. This is first time it was asked then :)

**Comment by alan2030** [comment]
> `customBottomWidget` is something appearing instead of the `Input`, like said in the documentation comment, for the channel view for example, where you can't reply. Original question I think is about buttons below messages? People actually achieved that using `bubbleBuilder` in #301, but there is no "official" way. Second time seeing this though, maybe should be added natively.

Oh yes, thats what I meant, but @henry2man option also sounds good.

**Comment by k00na** [comment]
Any updates on this? What's the current way of adding something like custom buttons bellow a message? 

We need custom answer buttons in our chat app, I can imagine this is a common requirement across chat bot apps.

**Comment by demchenkoalex** [comment]
@k00na This library became hard to maintain, so I am working on a complete re-design that will be more performant and customisable, that's why no updates. I will make sure to include this in the new requirements list.
